src/pkg/sypgp/sypgp.go

```diff
@@ -107,15 +108,28 @@ func printSignatures(entity *openpgp.Entity) error {
 	return nil
 }
 
+// AskQuestion prompts the user with a question and return the response
+func AskQuestion(format string, a ...interface{}) (string, error) {
+	fmt.Printf(format, a...)
+	scanner := bufio.NewScanner(os.Stdin)
+	scanner.Scan()
+	response := scanner.Text()
+	if err := scanner.Err(); err != nil {
+		return "", err
+	}
+	return response, nil
+}
+
 // DirPath returns a string describing the path to the sypgp home folder
 func DirPath() string {
 	user, err := user.GetPwUID(uint32(os.Getuid()))
 	if err != nil {
-		sylog.Errorf("could lookup user's real home folder %s\n", err)
-		return ""
+		sylog.Warningf("could not lookup user's real home folder %s\n", err)
+		sylog.Warningf("using current directory for %s\n", filepath.Join(".singularity", "sypgp"))
```

Comment:
Nitpick: logs shouldn't end with `\n`.

Reply:
Huh, so I just realized that sylog strips `\n` from the ending of any log messages, and then makes sure there is exactly one `\n` at the end. So there's a lot of code not doing this properly all over the rest of the codebase!

Reply:
Yeah, it's an idiom change from v2->v3, I'm sure these will be pervasive for a while. If you like, we could start linting for that 😉

Reply:
Yes and this could have been a leftover from the previous sylog call. sypgp was an early sylog adopter ;-) This is the right way to do it though and I believed I lobbied (with support) for that to happen also.

A general code cleanup is a good idea @bauerm97 !